### PR TITLE
oracled2: Ignore request not related to its validator

### LINKED ISSUE #1672: Make oracled2 ignore requests not related to its validator

### PR DESCRIPTION
**Pull request by perimeko**

### CODE REVIEW

File: chain/cmd/bandoracled2/handler.go [hunk 1]
```diff
@@ -56,7 +56,19 @@ func handleRequestLog(c *Context, l *Logger, log sdk.ABCIMessageLog) {
 	l = l.With("rid", id)
 	l.Info(":delivery_truck: Processing incoming request event")
```
Comment (sorawit) [inline chain/cmd/bandoracled2/handler.go]: Let's move this log to after the condition

File: chain/cmd/bandoracled2/handler.go [hunk 2]
```diff
@@ -56,7 +56,19 @@ func handleRequestLog(c *Context, l *Logger, log sdk.ABCIMessageLog) {
...
+	validatorAddr := GetEventValues(log, otypes.EventTypeRequest, otypes.AttributeKeyValidator)
+	isFoundValidator := false
+	for _, validator := range validatorAddr {
+		if validator == cfg.Validator {
+			isFoundValidator = true
+		}
+	}
+
+	if !isFoundValidator {
+		return
```
Comment (sorawit) [inline chain/cmd/bandoracled2/handler.go:70]: Let's have a debug log here. Choose your own emoji!

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandoracled2/handler.go [hunk 3]
```diff
@@ -53,10 +53,23 @@ func handleRequestLog(c *Context, l *Logger, log sdk.ABCIMessageLog) {
 		return
 	}
 
+	// Skip if not related to this validator
+	validatorAddr := GetEventValues(log, otypes.EventTypeRequest, otypes.AttributeKeyValidator)
```
Comment (sorawit) [inline chain/cmd/bandoracled2/handler.go]: Let's name this `validators`

File: chain/cmd/bandoracled2/handler.go [hunk 4]
```diff
@@ -53,10 +53,23 @@ func handleRequestLog(c *Context, l *Logger, log sdk.ABCIMessageLog) {
...
+	isFoundValidator := false
```
Comment (sorawit) [inline chain/cmd/bandoracled2/handler.go]: let's name thsi `hasMe`

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]
Looks great!

File: chain/cmd/bandoracled2/handler.go [hunk 5]
```diff
@@ -54,9 +54,23 @@ func handleRequestLog(c *Context, l *Logger, log sdk.ABCIMessageLog) {
 	}
 
 	l = l.With("rid", id)
+
+	// Skip if not related to this validator
+	validators := GetEventValues(log, otypes.EventTypeRequest, otypes.AttributeKeyValidator)
+	hasMe := false
+	for _, validator := range validators {
+		if validator == c.validator.String() {
+			hasMe = true
```
Comment (taobun) [inline chain/cmd/bandoracled2/handler.go:63]: Can break after found

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by Benzbeeb — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**